### ISSUE TITLE
Sharing widgets can lead to inconsistent UIs

### ISSUE THREAD

**Issue by romanofski** [purebred-mua/purebred#107]
The problem is that we currently have a 1:1 mapping between `Mode` and the widget which has focus. For drawing the UI, the code currently "hardcodes" additional UI elements to be drawn within this mode. The problem arises if we want to display the right set of widgets among the rest of focused widget.

For example: When listing threads we also want to be able to receive input to alter the search thus re-displaying a new result as well as changing the tags for a thread (setting tags on all messages). From a drawing perspective, all I know is that the current mode is eg `ManageTags` but no idea which other widgets I should draw along with it. Should it be the list of threads or the list of mails?

A possible work-around could be to check which list has a selected item, but that feels very hacky. So something more declarative is needed here.

**Comment by romanofski** [comment]
I also found that it's not only a problem with the UI, but also with our Keybindings. Again for managing tags. All I know when I pick up the user input in order to write it to the mail or thread is that the editor is focused. So here:

```Haskell
applyEditorMailTags :: AppState -> IO AppState
applyEditorMailTags s = case view asAppMode s of
  BrowseMail -> selectedItemHelper (asMailIndex . miListOfMails) s $ \m -> applyEditorMailTags' m s
  _ -> selectedItemHelper (asMailIndex . miListOfThreads) s $ \m -> applyEditorMailTags' m s
```

will actually not do what I thought it will do. The pattern match matches on the catch all branch, since the current app mode is `ManageTags` therefore setting all tags to the mails in the currently selected thread.

**Comment by romanofski** [comment]
Had a chat with @frasertweedale over this and we concluded that it's better to go with more `Mode`s at this point. For this particular issue, we'd introduce a `ManageMailTags` mode and a `ManageThreadTags` mode.

**Comment by romanofski** [comment]
I'm closing this for now since it's addressed by #103
